### PR TITLE
Surface the use of regions in doc samples

### LINKED ISSUE #9620: Surface the use of snippets in the sample download section
See: https://github.com/aspnet/Docs/pull/9544#issuecomment-439299077

cc: @rswetnam

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: ad8c13e7-78e6-cf9a-e7cb-3215df5fd1c4
* Version Independent ID: a3c1d2a6-625a-3409-120d-8372230a8196
* Content: [Introduction to ASP.NET Core](https://docs.microsoft.com/en-us/aspnet/core/?view=aspnetcore-2.1#how-to-download-a-sample)
* Content Source: [aspnetcore/index.md](https://github.com/aspnet/Docs/blob/master/aspnetcore/index.md)
* Product: **aspnet-core**
* GitHub Login: @Rick-Anderson
* Microsoft Alias: **riande**

### PR DESCRIPTION
**Pull request by guardrex**
Fixes #9620 

Thanks @rswetnam

### CODE REVIEW

File: aspnetcore/index.md [hunk 1]
```diff
@@ -87,6 +89,27 @@ To change the sample to run the `ExpandDefault` scenario, define the `ExpandDefa
 
 For more information on using [C# preprocessor directives](/dotnet/csharp/language-reference/preprocessor-directives/) to selectively compile sections of code, see [#define (C# Reference)](/dotnet/csharp/language-reference/preprocessor-directives/preprocessor-define) and [#if (C# Reference)](/dotnet/csharp/language-reference/preprocessor-directives/preprocessor-if).
 
+### Sample regions
```
Comment (tdykstra) [inline aspnetcore/index.md]: Regions in sample code

File: aspnetcore/index.md [hunk 2]
```diff
@@ -71,6 +71,8 @@ Many of the articles and tutorials include links to sample code.
 1. Unzip the *Docs-master.zip* file.
 1. Use the URL in the sample link to help you navigate to the sample directory.
 
+### Sample preprocessor directives
```
Comment (tdykstra) [inline aspnetcore/index.md]: Preprocessor directives in sample code

File: aspnetcore/index.md [hunk 3]
```diff
@@ -71,6 +71,8 @@ Many of the articles and tutorials include links to sample code.
...
+
 To demonstrate multiple scenarios, sample apps make use of the `#define` and `#if-#else/#elif-#endif` C# statements to selectively compile and run different sections of sample code. For those samples that make use of this approach, set the `#define` statement at the top of the C# files to the symbol associated with the scenario that you want to run. A sample may require you to set the symbol at the top of multiple files in order to run a scenario.
```
Comment (tdykstra) [inline aspnetcore/index.md]: make use of --> use

A sample may require you to--> You might have to

File: aspnetcore/index.md [hunk 4]
```diff
@@ -87,6 +89,27 @@ To change the sample to run the `ExpandDefault` scenario, define the `ExpandDefa
 
 For more information on using [C# preprocessor directives](/dotnet/csharp/language-reference/preprocessor-directives/) to selectively compile sections of code, see [#define (C# Reference)](/dotnet/csharp/language-reference/preprocessor-directives/preprocessor-define) and [#if (C# Reference)](/dotnet/csharp/language-reference/preprocessor-directives/preprocessor-if).
 
+### Sample regions
+
+Documentation sample apps contain sections of code surrounded by [#region](/dotnet/csharp/language-reference/preprocessor-directives/preprocessor-region) and [#end-region](/dotnet/csharp/language-reference/preprocessor-directives/preprocessor-endregion) C# statements. The documentation build system injects these regions into the rendered documentation topics.
```
Comment (tdykstra) [inline aspnetcore/index.md]: Documentation sample apps --> Some sample apps

File: aspnetcore/index.md [hunk 5]
```diff
@@ -87,6 +89,27 @@ To change the sample to run the `ExpandDefault` scenario, define the `ExpandDefa
 
 For more information on using [C# preprocessor directives](/dotnet/csharp/language-reference/preprocessor-directives/) to selectively compile sections of code, see [#define (C# Reference)](/dotnet/csharp/language-reference/preprocessor-directives/preprocessor-define) and [#if (C# Reference)](/dotnet/csharp/language-reference/preprocessor-directives/preprocessor-if).
 
+### Sample regions
+
+Documentation sample apps contain sections of code surrounded by [#region](/dotnet/csharp/language-reference/preprocessor-directives/preprocessor-region) and [#end-region](/dotnet/csharp/language-reference/preprocessor-directives/preprocessor-endregion) C# statements. The documentation build system injects these regions into the rendered documentation topics.  
+
+Region names usually contain the word "snippet." The following example shows a region named `snippet_FilterInCode`:
```
Comment (tdykstra) [inline aspnetcore/index.md]: After this, show the line of code in markdown that pulls in this snippet.

File: aspnetcore/index.md [hunk 6]
```diff
@@ -87,6 +89,27 @@ To change the sample to run the `ExpandDefault` scenario, define the `ExpandDefa
...
+WebHost.CreateDefaultBuilder(args)
+    .UseStartup<Startup>()
+    .ConfigureLogging(logging =>
+        logging.AddFilter("System", LogLevel.Debug)
+            .AddFilter<DebugLoggerProvider>("Microsoft", LogLevel.Trace))
+            .Build();
+#endregion
+```
+
+You may safely ignore (or remove) the `#region` and `#end-region` statements that surround the code. Don't alter the code within these statements.
```
Comment (tdykstra) [inline aspnetcore/index.md]: They should feel free to monkey around with samples ... consider removing "(or remove)" and "Don't alter ..."

File: aspnetcore/index.md (hunk 5 above)
Comment (guardrex) [reply aspnetcore/index.md]: I didn't do it for two reasons:

* It blows our 85-ish char limit to show it.
* This section is just about why they're there. If they want to know more, the linked content takes care of it.

I'll put in a *fake* markdown line (shorter) to show it ... try to keep it <85 chars.

File: aspnetcore/index.md (hunk 6 above)
Comment (guardrex) [reply aspnetcore/index.md]: :smile: ... I was trying to ward off *removing the entire snippet* by accident. Let me see what I can do on the next commit to improve this.

**Comment by guardrex** [comment]
@tdykstra kk ... see how that looks. It's correct that they can either ignore *or remove* the region lines, so I left that remark there.

**Comment by tdykstra** [comment]
Looks good, thanks. Glad you were able to squeeze in the line referring to a snippet.

**Pull request merged**